### PR TITLE
added URINormalizer

### PR DESCRIPTION
**Pull request by ymatsuda**
- does general URI normalization
  - lowercasing protocol and host name
  - relative path -> absolute path
  - escape normalization
- query string
  - sorting parameters by name
  - removal of well known session parameters
  - removal of well known tracking parameters
- error recovery for malformed escapes
- whenever a part of normalization fails, it logs errors and fallback to the original

### CODE REVIEW

File: shoebox/app/com/keepit/common/net/URINormalizer.scala [hunk 1]
```diff
@@ -0,0 +1,139 @@
+package com.keepit.common.net
+
+import com.keepit.common.logging.Logging
+import scala.collection.immutable.SortedMap
+import scala.collection.immutable.SortedSet
+
+object URINormalizer extends Logging {
+  
+  val stopParams = Set("jsessionid", "phpsessid", "aspsessionid", "utm_source", "utm_medium", "utm_campaign", "utm_term", "utm_content")
```
Comment (eishay) [inline shoebox/app/com/keepit/common/net/URINormalizer.scala:9]: i think you should add zenid as well http://rubikin.com/2009/08/17/zencart-seo-problem-is-your-site-indexed/
and sid http://www.unicon.net/node/497

**Comment by eishay** [comment]
cool! like the unapply :smiley:

File: shoebox/app/com/keepit/common/net/URINormalizer.scala (hunk 1 above)
Comment (ymatsuda) [inline shoebox/app/com/keepit/common/net/URINormalizer.scala:9]: I will add "zenid". But "sid" sounds very generic name. I am concerned about side effects.

Comment (eishay) [inline shoebox/app/com/keepit/common/net/URINormalizer.scala:9]: ok, lets log all normalizations we're doing and grep it from time to time
from the log to inspect the results we see in real life.

**Pull request merged**